### PR TITLE
Fix tensorrt model serialization and caching issue

### PR DESCRIPTION
**Pull request by DVEfremov**

### CODE REVIEW

File: program/tensorrt-time/tensorrt-time.cpp [hunk 1]
```diff
@@ -229,7 +229,22 @@ void convertCaffeToTensorRT(
     parser->destroy();
 
     // Serialize the engine, then shut everything down.
-  //  engine->serialize(output_stream);
+#if NV_TENSORRT_MAJOR > 1
+	nvinfer1::IHostMemory* serMem = engine->serialize();
+
+	if( !serMem )
+	{
+		printf(LOG_GIE "failed to serialize CUDA engine\n");
```
Comment (psyhtest) [inline program/tensorrt-time/tensorrt-time.cpp]: I know this program mixes `printf` and `std::cout`, but can we make it more uniform? How about we only use `printf` in `main` and `std::cout << "\n[tensorrt-time] - ` elsewhere?

File: program/tensorrt-time/tensorrt-time.cpp [hunk 2]
```diff
@@ -429,8 +445,14 @@ int main(int argc, char** argv)
                 logger);
             std::cout << "\n[tensorrt-time] - storing...";
             std::ofstream tensorrt_model_cache_store(tensorrt_model_cache_path);
+
+            if (! tensorrt_model_cache_store) {
+                printf("tensorrt_model_cache_store could not be opened\n");
```
Comment (psyhtest) [inline program/tensorrt-time/tensorrt-time.cpp]: How about:
```
std::cout << "\n[tensorrt-time] Could not open cache for writing\n";
exit(EXIT_FAILURE);
```

File: program/tensorrt-time/tensorrt-time.cpp [hunk 3]
```diff
@@ -451,6 +473,7 @@ int main(int argc, char** argv)
 	// instead, read the stringstream into a memory buffer and pass that to TRT.
 	tensorrt_model_stream.seekg(0, std::ios::end);
 	const int modelSize = tensorrt_model_stream.tellg();
+        printf("\n modelSize=%d\n", modelSize);
```
Comment (psyhtest) [inline program/tensorrt-time/tensorrt-time.cpp]: Drop?

**Comment by psyhtest** [comment]
Line 484 or so:
```
		printf("failed to allocate %i bytes to deserialize model\n", modelSize);
		return 0;
```
Shouldn't this be something like:
```
std::cout << "[tensorrt-time] Failed to allocate " << modelSize << " bytes to deserialize model!\n";
exit(EXIT_FAILURE);
```

**Comment by DVEfremov** [comment]
I've fixed review comments and merging PR as we've discussed

**Pull request merged**